src/api/app/controllers/status/checks_controller.rb

```diff
@@ -1,71 +1,58 @@
 class Status::ChecksController < ApplicationController
-  before_action :require_project
-  before_action :require_repository
-  before_action :require_checkable, only: [:index, :show, :destroy, :update]
-  before_action :require_or_initialize_checkable, only: :create
-  before_action :require_check, only: [:show, :destroy, :update]
-  before_action :set_xml_check, only: [:create, :update]
-  skip_before_action :require_login, only: [:show, :index]
+  before_action :set_xml_check
+  before_action :require_checkable
+  before_action :require_status_report
+  before_action :require_check
   after_action :verify_authorized
 
-  # GET /projects/:project_name/repositories/:repository_name/repository_publishes/:repository_publish_build_id/checks
-  def index
-    @checks = @checkable.checks
-    @missing_checks = @checkable.missing_checks
-    authorize @checks
-  end
-
-  # GET /projects/:project_name/repositories/:repository_name/repository_publishes/:repository_publish_build_id/checks/:id
-  def show
-    authorize @check
-  end
+  # PUT /status_reports/published/:project_name/:repository_name/reports/:uuid
+  # PUT /status_reports/requests/:number
+  def update
+    authorize @status_report
 
-  # POST /projects/:project_name/repositories/:repository_name/repository_publishes/:repository_publish_build_id/checks
-  def create
-    @xml_check[:checkable] = @checkable
-    @check = Status::Check.new(@xml_check)
-    authorize @check
-    if @check.save
-      render :show
+    if @check
+      @check.assign_attributes(@xml_check)
     else
-      render_error(status: 422, errorcode: 'invalid_check', message: "Could not save check: #{@check.errors.full_messages.to_sentence}")
+      @xml_check[:status_report] = @status_report
+      @check = Status::Check.new(@xml_check)
     end
-  end
 
-  # PATCH /projects/:project_name/repositories/:repository_name/repository_publishes/:repository_publish_build_id/checks/:id
-  def update
-    authorize @check
-    if @check.update(@xml_check)
+    if @check.save
       render :show
     else
       render_error(status: 422, errorcode: 'invalid_check', message: "Could not save check: #{@check.errors.full_messages.to_sentence}")
     end
   end
 
-  # DELETE /projects/:project_name/repositories/:repository_name/repository_publishes/:repository_publish_build_id/checks/:id
-  def destroy
-    authorize @check
-    if @check.destroy
-      render_ok
+  private
+
+  def require_status_report
+    if params[:report_uuid]
+      @status_report = @checkable.status_reports.find_or_initialize_by(uuid: params[:report_uuid])
     else
-      render_error(status: 422, errorcode: 'invalid_check', message: "Could not delete check: #{@check.errors.full_messages.to_sentence}")
+      @status_report = @checkable.status_reports.first_or_initialize
     end
   end
 
-  private
+  def require_checkable
+    set_repository || set_bs_request
+    return if @checkable
+    raise ActiveRecord::RecordNotFound
```

Comment:
As a User I don't know what is missing. So I can't fix my API call. What is the problem with having a chain of `before_actions`?